learn/core_concepts/documents.md

```diff
@@ -1,26 +1,58 @@
+
+If a field contains an object, Meilisearch flattens it during indexation using dot notation and brings the object's keys and values to the root level of the document itself. This flattened object is an intermediary representation and you will get the original structure upon search. You can read more about this in our [dedicated guide](/learn/advanced/datatypes.md#objects).
+
+You can apply [ranking rules](/learn/core_concepts/relevancy.md#ranking-rules) to some fields. For example, you may decide recent movies should be more relevant than older ones. You can also configure how Meilisearch handles certain fields at an [index level](/learn/configuration/settings.md) in the settings.
+
+### Displayed and searchable fields
+
+By default, all fields in a document are both displayed and searchable. You can modify this behavior using the [update settings endpoint](/reference/api/settings.md#update-settings), or the respective update endpoints for [displayed attributes](/reference/api/displayed_attributes.md#update-displayed-attributes), and [searchable attributes](/reference/api/searchable_attributes.md#update-searchable-attributes) so that a field is:
+
+- Searchable but not displayed
+- Displayed but not searchable
+- Neither displayed nor searchable
+
+In the latter case, the field will be completely ignored during search. However, it will still be [stored](/learn/configuration/displayed_searchable_attributes.md#data-storing) in the document.
+
+## Primary field
+
+The primary field is a special field that must be present in all documents. Its attribute is the [primary key](/learn/core_concepts/primary_key.md#primary-key-2) and its value is the [document id](/learn/core_concepts/primary_key.md#document-id). If you try to [index a document](/learn/getting_started/quick_start.md#add-documents) that's incorrectly formatted, missing a primary key, or possessing the wrong primary key for a given index, it will cause an error and no documents will be added.
+
+To learn more, refer to the [primary key explanation](/learn/core_concepts/primary_key.md).
+
+## Upload
+
+By default, Meilisearch limits the size of all payloads—and therefore document uploads—to 100MB. You can [change the payload size limit](/learn/configuration/instance_options.md#payload-limit-size) at runtime using the `http-payload-size-limit` option.
+
+**Meilisearch uses a lot of RAM when indexing documents**. Be aware of your [RAM availability](/resources/faq.md#what-are-the-recommended-requirements-for-hosting-a-meilisearch-instance) as you increase your batch size as this could cause Meilisearch to crash.
+
+When using the [add new documents endpoint](/reference/api/documents.md#add-or-update-documents), all documents must be sent in an array even if there is only one document.
 
 ### Dataset format
 
-You can provide your dataset in the following formats:
+Meilisearch accepts datasets in the following formats:
 
 - [JSON](#json)
```

Comment:
Are these misaligned on the file itself or is it just an artefact of github's interface?

Reply:
It's GitHub, they're aligned:

![Screen Shot 2022-06-07 at 12 15 09](https://user-images.githubusercontent.com/90181761/172330925-a5ee92a8-1d78-41b8-bb10-b98b015a10da.png)